_source/_posts/2021-01-20-reactive-java-microservices.adoc

```diff
@@ -0,0 +1,545 @@
+    pointsSearchRepository.save(result);
+    return ResponseEntity
+        .created(new URI("/api/points/" + result.getId()))
+        .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, result.getId().toString()))
+        .body(result);
+}
+----
+
+The same functionality when implemented with Spring WebFlux returns a `Mono` and uses a more functional, streaming style to avoid blocking.
+
+[source,java]
+----
+/**
+ * {@code POST  /points} : Create a new points.
+ *
+ * @param points the points to create.
+ * @return the {@link ResponseEntity} with status {@code 201 (Created)} and with body the new points, or with status {@code 400 (Bad Request)} if the points has already an ID.
+ * @throws URISyntaxException if the Location URI syntax is incorrect.
+ */
+@PostMapping("/points")
+public Mono<ResponseEntity<Points>> createPoints(@Valid @RequestBody Points points) throws URISyntaxException {
+    log.debug("REST request to save Points : {}", points);
+    if (points.getId() != null) {
+        throw new BadRequestAlertException("A new points cannot already have an ID", ENTITY_NAME, "idexists");
+    }
+    return pointsRepository
+        .save(points)
+        .flatMap(pointsSearchRepository::save)
+        .map(
+            result -> {
+                try {
+                    return ResponseEntity
+                        .created(new URI("/api/points/" + result.getId()))
+                        .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, result.getId().toString()))
+                        .body(result);
+                } catch (URISyntaxException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        );
+}
+----
+
+In this guide, I'll show you how to create a reactive microservices architecture with Spring Boot, Spring Cloud Gateway, Spring Cloud Config, Spring WebFlux, Java, and JHipster.
+
+image::{% asset_path 'blog/reactive-java-microservices/reactive-java-microservices.png' %}[alt=Reactive Java Microservices,width=800,align=center]
+
+**Prerequisites**
+
+* https://adoptopenjdk.net/[Java 11]+
+* https://docs.docker.com/engine/install/[Docker]
+
+You can find the completed source code for this example on GitHub, in the https://github.com/oktadeveloper/java-microservices-examples/tree/main/reactive-jhipster[oktadeveloper/java-microservices-examples repository].
+
+[source,shell]
+----
+git clone https://github.com/oktadeveloper/java-microservices-examples.git
+cd java-microservices-examples/reactive-jhipster
+----
+
+== Build a Reactive Java Microservices Architecture
+
+https://jhipster.tech[JHipster] is an open-source project that started as an application generator, based on Yeoman. It initially allowed you to generate AngularJS and Spring apps. In the last five years, it's become a platform with extensibility, grown a thriving community, and has saved developers many hours of pain and frustration.
```

Comment:
Add link to Yeomen? I only kinda know what Yeomen is because of JHipster

Reply:
Will do.